packages/data-sql/src/utils/number-generator.ts

```diff
@@ -0,0 +1,12 @@
+export type NumberGenerator = Generator<number, never, never>;
+
+/**
+ * Generator function to generate parameter indices.
+ */
+export function* numberGenerator(): NumberGenerator {
```

Comment:
since this function is so tiny and it's very closely related to the index generators, shall we move it to the `create-index-generator` file? 🤔

Reply:
Maybe we should add a `utils/utils` folder for utils which are used by utils 🥴

Reply:
> since this function is so tiny and it's very closely related to the index generators, shall we move it to the create-index-generator file? 🤔

I'd prefer keeping it one exported function per file 👍🏻 Having them next to eachother is fine

Reply:
alright! now that it's within `converter/utils` it also seems more appropriate to me having it in a separate file :)